Lib/test/test_builtin.py

```diff
@@ -1511,6 +1513,126 @@ def test_construct_singletons(self):
             self.assertRaises(TypeError, tp, 1, 2)
             self.assertRaises(TypeError, tp, a=1, b=2)
 
+
+class TestBreakpoint(unittest.TestCase):
```

Comment:
Maybe you should add a setUp() method using self.addCleanup() to save/restore sys.breakpointhook value, instead of doing that in each test.

Reply:
Yes, this is a great suggestion, coupled with the observation below that the test suite could be run with `PYTHONBREAKPOINT` or `sys.breakpointhook` set, we really need the `setUp()` to clean the slate for the subsequent tests.

I'm going to use an `ExitStack()` to do this in the `setUp()` with a `addCleanup()` to close the context manager and restore the original state.